homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,47 @@
+#include "compression.hpp"
+
+std::vector<int> vectorCopy(std::vector<int> input){
+    return input;
+}
+
+std::vector<std::pair<uint8_t,uint8_t>>  compressGrayscale(std::array<std::array<uint8_t,width>,height> bitmap) {
+    std::vector<std::pair<uint8_t,uint8_t>> compressed;
+    compressed.reserve(width * height);
+    for (const auto& line : bitmap) {
+        uint8_t counter = 1;
+        for (auto it = line.begin(); it != line.end(); it++) {
+            if (*it != *std::next(it) || it == line.end() - 1) {
+                compressed.push_back(std::make_pair(*it, counter));
+                counter = 1;
+            } else {
+                counter++;
+            }
+        }
+    }
+    return compressed;
+}
+
+std::array<std::array<uint8_t,width>,height>  decompressGrayscale(std::vector<std::pair<uint8_t,uint8_t>> compressed) {
+    std::array<std::array<uint8_t,width>,height> decompressed;
+    /*
+    Transforms vector of pairs(char value,char count) into a 2D-array of char
+    */
```

Comment:
Comment is not necessary. Mateusz said good code don't need too much explantation in comment, I thinking yours is good